drizzlepac/haputils/catalog_utils.py

```diff
@@ -248,38 +250,48 @@ def compute_background(self, box_size, win_size,
         bkg = None
         is_zero_background_defined = False
 
-        # Make a local copy of the data(image) being processed in order to reset any
-        # data values which equal nan (e.g., subarrays) to zero.
-        imgdata = np.nan_to_num(self.data, copy=True, nan=0.0)
```

Comment:
Will altering this line result in nans in the image?

Reply:
Took some time to double-check!  There will still be nans in the "image", but the masks (noting where data should or should not be used) are employed when performing important computations.